client/web/src/enterprise/code-monitoring/components/CodeMonitorForm.tsx

```diff
@@ -94,19 +94,22 @@ export const CodeMonitorForm: React.FunctionComponent<CodeMonitorFormProps> = ({
             (submit: Observable<React.FormEvent<HTMLFormElement>>) =>
                 submit.pipe(
                     tap(event => event.preventDefault()),
-                    mergeMap(() =>
-                        onSubmit(currentCodeMonitorState).pipe(
-                            startWith(LOADING),
-                            catchError(error => [asError(error)]),
-                            tap(successOrError => {
-                                if (!isErrorLike(successOrError) && successOrError !== LOADING) {
-                                    history.push('/code-monitoring')
-                                }
-                            })
-                        )
-                    )
+                    mergeMap(() => {
+                        if (formCompletion.actionCompleted && formCompletion.triggerCompleted) {
+                            return onSubmit(currentCodeMonitorState).pipe(
+                                startWith(LOADING),
+                                catchError(error => [asError(error)]),
+                                tap(successOrError => {
+                                    if (!isErrorLike(successOrError) && successOrError !== LOADING) {
+                                        history.push('/code-monitoring')
+                                    }
+                                })
+                            )
+                        }
+                        return of(undefined)
+                    })
```

Comment:
```suggestion
                    filter(() => formCompletion.actionCompleted && formCompletion.triggerCompleted),
                    mergeMap(() => onSubmit(currentCodeMonitorState).pipe(
                            startWith(LOADING),
                            catchError(error => [asError(error)]),
                            tap(successOrError => {
                                if (!isErrorLike(successOrError) && successOrError !== LOADING) {
                                    history.push('/code-monitoring')
                                }
                            })
                        )
                 )
```

Nit: maybe a bit more explicit?

On another note, I find it weird to filter or conditionally map in an Observable pipe based on data external to the pipe, but cannot come up with a more 'philosophically correct' solution that doesn't take a readability hit.

Reply:
Yeah I also felt this was an anti-pattern but I wasn't sure if there was an alternative here... I couldn't come up with one that seemed reasonable.